src/webgpu/util/f32_interval.ts

```diff
@@ -9,6 +9,14 @@ import {
   oneULP,
 } from './math.js';
 
+/**
+ * Representation of bounds for an interval as an array with either one or two
+ * elements. Single element indicates that the interval is a single point. For
+ * two elements, the first is the lower bound of the interval and the second is
+ * the upper bound.
+ */
+export type IntervalBounds = [number] | [number, number];
```

Comment:
I can see your logic for this from the test specification perspective.

The main reason I haven't previous done this, is because that would be mean that bounds would not be guaranteed to be an array, so things like rest parameters (...foo), functional operations (.map, .forEach), etc don't work out of the box, so there would need to be type coercion/conversion going on somewhere or different logic for a number vs an array of numbers.

The bounds() return value could be changed to a [number, number] like we had before, but then we are no longer guaranteed that the entries are unique, so there has to be filtering through Set to guarantee uniqueness at various places.

bounds() could instead return [number] | [number, number] to resolve the uniqueness question, but then the output of bounds can not be directly fed into the F32Interval constructor without checking the length and unpacking, since we cannot just use ...

I could make the type be number | [number] | [number, number] to resolved the bounds into constructor situation, but now  we still have that the might not be an array problem in various places.

We could split the type between the tests and the implementation, letting the tests use number | [number, number], but that still has the it might not be an array when passing it in.

I think this is a situation of trade offs and picking the solution that is most palatable. I have gone with adding a little bit of cruft in the form of extra []s in data for the unittests, to allow for the actual implementation and test code being cleaner.

If you have an alternates to avoid this whole circular/chicken-egg problem I am interested in hearing them. I have gone around this circle a few times over the life of the code and I think this is tradeoff I am currently satisfied with.

Reply:
This is fine as-is. Was just curious. LGTM

Reply:
n/p, it was good to revisit why it was like this to make sure there wasn't an obvious other way to do it that didn't have a downside.